### PR TITLE
Improved content picker for piping

### PR DESCRIPTION
**Pull request by pricea1**
### What is the context of this PR?
New design for content picker

### How to review 
1. Ensure piping and "select an answer" in calcuated summary page still works.
2. Ensure styling and functionality match prototype: [https://prototype-picker--author-app.netlify.com/](https://prototype-picker--author-app.netlify.com/)
3. Should comply with acceptance criteria: [Answer picker](https://trello.com/c/b9jLEjJN/1025-improve-content-picker-piping-an-answer-6) and [Metadata picker](https://trello.com/c/AwyAOkvA/1216-improve-content-picker-piping-metadata-4)

### CODE REVIEW

**Comment by pricea1** [comment]
[Improve content picker - piping an answer (6)](https://trello.com/c/b9jLEjJN/1025-improve-content-picker-piping-an-answer-6)

**Comment by pricea1** [comment]
[Improve content picker - piping metadata (2)](https://trello.com/c/AwyAOkvA/1216-improve-content-picker-piping-metadata-2)

File: eq-author/src/components/ContentPickerv2/AnswerPicker.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          <SectionMenu data={data} {...otherProps} />
+        )}
+      </MenuContainer>
+    </>
+  );
+};
+
+AnswerPicker.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author/src/components/ContentPickerv2/AnswerPicker.js]: Now changed

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+const SubMenu = ({
+  data,
+  onSelected,
+  isSelected,
+  isDisabled,
+  ...otherProps
+}) => {
+  const onEnterUp = (event, item) => {
+    if (event.keyCode === 13) {
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author/src/components/ContentPickerv2/Menu.js]: Keyboard accessibility now added to section menu

File: eq-author/src/components/ContentPickerv2/MetadataPicker.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+const MetaDataItemList = styled.ul`
+  display: table-row-group;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`;
+
+const MetaDataPicker = ({ data, isSelected, onSelected }) => {
+  const onEnterUp = (event, item) => {
+    if (event.keyCode === 13) {
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author/src/components/ContentPickerv2/MetadataPicker.js]: This should already have been working? Are you getting an issue?

File: eq-author/src/components/ContentPickerv2/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  isOpen,
+  startingSelectedAnswers,
+  multiselect,
+  onSubmit,
+}) => {
+  const [selectedItems, setSelectedItems] = useState(startingSelectedAnswers);
+
+  useEffect(() => {
+    setSelectedItems(startingSelectedAnswers);
+  }, [startingSelectedAnswers]);
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author/src/components/ContentPickerv2/index.js]: Not needed so now removed.

File: eq-author/src/components/ContentPickerv2/MetadataPicker.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/components/ContentPickerv2/MetadataPicker.js]: Cant seem to tab into the menu items themselves at the moment.
ezgif-1-61bfe6c54525

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+          );
+        })}
+    </MenuItemList>
+  );
+};
+
+Menu.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```
Comment (SamGodwin2) [inline eq-author/src/components/ContentPickerv2/Menu.js]: Sorry missed these earlier, can we use the custom-prop type here as well, helps to understand the component at a glance.

File: eq-author/src/components/ContentPickerv2/Menu.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    <div key={section.id}>
+      <SectionTitle>{section.displayName}</SectionTitle>
+      <SubMenu data={section.pages} {...otherProps} />
+    </div>
+  ));
+
+FlatSectionMenu.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```
Comment (SamGodwin2) [inline eq-author/src/components/ContentPickerv2/Menu.js]: Same as above with custom-prop types

File: eq-author/src/components/ContentPickerv2/MetadataPicker.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        </ScrollPane>
+      </MenuContainer>
+    </>
+  );
+};
+
+MetaDataPicker.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```
Comment (SamGodwin2) [inline eq-author/src/components/ContentPickerv2/MetadataPicker.js]: Same here for metadata

File: eq-author/src/components/ContentPickerv2/SectionMenu.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        </ScrollPane>
+      </Column>
+    </Columns>
+  );
+};
+
+SectionMenu.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```
Comment (SamGodwin2) [inline eq-author/src/components/ContentPickerv2/SectionMenu.js]: Think this one is sections again?

File: eq-author/src/components/ContentPickerv2/index.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+        </ButtonGroup>
+      </ModalFooter>
+    </StyledModal>
+  );
+};
+
+ContentPicker.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```
Comment (SamGodwin2) [inline eq-author/src/components/ContentPickerv2/index.js]: Can we use `PropTypes.oneOfType([customPropTypes.Metadata, customPropTypes.Answer])` here

Comment (pricea1) [reply eq-author/src/components/ContentPickerv2/index.js]: All propTypes changes sorted

**Comment by heofs** [comment]
I sometimes get this error message when clicking on the section navigation in the content picker. 
```
Warning: Failed prop type: Invalid prop `firstSelectedItemId` of type `boolean` supplied to `SectionMenu`, expected `string`.
```

**Comment by pricea1** [comment]
> I sometimes get this error message when clicking on the section navigation in the content picker.
> 
> ```
> Warning: Failed prop type: Invalid prop `firstSelectedItemId` of type `boolean` supplied to `SectionMenu`, expected `string`.
> ```

This happened when content picker opened with multiple sections to choose from but no answers selected.

Now fixed.

File: eq-author/src/components/ContentPickerv2/index.js [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+import Modal, { CloseButton } from "components/modals/Modal";
+import Button from "components/buttons/Button";
+import ButtonGroup from "components/buttons/ButtonGroup";
+
+import { colors } from "constants/theme";
+
+import { ANSWER, METADATA } from "components/ContentPickerSelect/content-types";
+import AnswerPicker from "./AnswerPicker";
+import MetadataPicker from "./MetadataPicker";
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.js]: Could you order the imports here a little better?

File: eq-author/src/components/ContentPickerv2/index.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+  position: absolute;
+  right: 0.5em;
+  top: 2em;
+  bottom: 0;
+  margin: auto;
+  align-items: center;
+`;
+
+const StyledModal = styled(Modal)`
+  .Modal {
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.js]: Is `.Modal` referencing itself? If so, could you do `& {...` or maybe just remove that selector?

File: eq-author/src/components/ContentPickerv2/index.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+  startingSelectedAnswers,
+  multiselect,
+  onSubmit,
+}) => {
+  const [selectedItems, setSelectedItems] = useState(startingSelectedAnswers);
+
+  const isSelected = item => {
+    return (
+      selectedItems.findIndex(selectedItem => selectedItem.id === item.id) !==
+      -1
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.js]: Can you shorten this with a single expression return?

File: eq-author/src/components/ContentPickerv2/index.test.js [hunk 13]
```diff
@@ -0,0 +1,424 @@
...
+    props = { data, onClose, onSubmit, startingSelectedAnswers };
+  });
+
+  const renderContentPicker = () => render(<ContentPicker {...props} isOpen />);
+
+  it("should not render picker by default", () => {
+    const { queryByTestId } = renderContentPicker();
+
+    const modalHeader = queryByTestId("content-picker-empty");
+    expect(modalHeader).not.toBeNull();
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.test.js]: Can we instead check for truthy here?

File: eq-author/src/components/ContentPickerv2/index.test.js [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+        ...props,
+        contentType: ANSWER,
+      };
+    });
+
+    it("should render answer picker when specified", () => {
+      const { getByText } = renderContentPicker();
+
+      const modalHeader = getByText("Select an answer");
+      expect(modalHeader).not.toBeNull();
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.test.js]: Same here.

File: eq-author/src/components/ContentPickerv2/index.test.js [hunk 15]
```diff
@@ -0,0 +1,424 @@
...
+        data,
+        contentType: METADATA,
+      };
+    });
+
+    it("should render metadata picker when specified", () => {
+      const { getByText } = renderContentPicker();
+
+      const modalHeader = getByText("Select metadata");
+      expect(modalHeader).not.toBeNull();
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.test.js]: Maybe change to `toBeTruthy()`?

File: eq-author/src/components/ContentPickerv2/index.test.js [hunk 16]
```diff
@@ -0,0 +1,424 @@
...
+
+      fireEvent.click(confirmButton);
+
+      expect(meta1Item).toHaveAttribute("aria-selected", "false");
+      expect(meta2Item).toHaveAttribute("aria-selected", "true");
+
+      expect(onSubmit).toHaveBeenCalledWith(meta2);
+    });
+
+    it("should unselect 1selected item", () => {
```
Comment (heofs) [inline eq-author/src/components/ContentPickerv2/index.test.js]: `1selected`?

File: eq-author/src/components/RichTextEditor/PipingMenu.js [hunk 17]
```diff
@@ -82,39 +70,45 @@ export class Menu extends React.Component {
...
       <React.Fragment>
         <MenuButton
-          {...buttonProps}
-          disabled={isDisabled}
-          onClick={this.handleButtonClick}
+          title="Pipe value"
+          disabled={loading || disabled || isEmpty(answerData)}
+          onClick={() => {
+            this.handleButtonClick(ANSWER);
+          }}
```
Comment (heofs) [inline eq-author/src/components/RichTextEditor/PipingMenu.js]: Could you shorten this with a single expression return?

File: eq-author/src/components/RichTextEditor/PipingMenu.js [hunk 18]
```diff
@@ -82,39 +70,45 @@ export class Menu extends React.Component {
...
-          metadataData={metadataData}
+        >
+          <IconPiping />
+        </MenuButton>
+        <MenuButton
+          title="Pipe metadata"
+          disabled={loading || disabled || isEmpty(metadataData)}
+          onClick={() => {
+            this.handleButtonClick(METADATA);
+          }}
```
Comment (heofs) [inline eq-author/src/components/RichTextEditor/PipingMenu.js]: Same with this one.

**Pull request merged**